shepherd/Codegen.py

```diff
@@ -0,0 +1,368 @@
+import random
+
+class Codegen:
+
+    def __init__(self, rfids, rand_seed=None):
+        '''
+        Manages the generated code and rfid-answer pairs.
+        `rfids` must be an iterable of RFID ints.
+        Duplicate RFIDs in the provided iterable are ignored.
+        '''
+
+        # Seed our local RNG
+        self._random = random.Random(rand_seed)
+        rfids = set(rfids) # Ignore duplicates
```

Comment:
I actually don't think we need to handle cases where we have duplicate RFIDs (if anything, if there exist duplicates we've already screwed up). A set might still be the easiest thing to use, but is not strictly necessary.